test/k8sT/istio.go

```diff
@@ -0,0 +1,325 @@
+
+	var (
+		// istioSystemNamespace is the default namespace into which Istio is
+		// installed.
+		istioSystemNamespace = "istio-system"
+
+		// istioYAMLPath is the istio-cilium.yaml file generated following the
+		// instructions in Documentation/gettingstarted/istio.rst to setup
+		// Istio 1.0.0. mTLS is enabled.
+		istioYAMLPath = helpers.ManifestGet("istio-cilium.yaml")
+
+		// istioServiceNames is the subset of Istio services in the Istio
+		// namespace that are accessed from sidecar proxies.
+		istioServiceNames = []string{
+			// "istio-galley",
+			// "istio-egressgateway",
+			// "istio-ingressgateway",
+			// "istio-policy",
+			// "istio-telemetry",
+			"istio-statsd-prom-bridge",
+			"istio-pilot",
+			// "prometheus",
+			// "istio-citadel",
+			// "istio-sidecar-injector",
+		}
+
+		// wgetCommand is the command used in this test because the Istio apps
+		// do not provide curl.
+		wgetCommand = fmt.Sprintf("wget --tries=2 --connect-timeout %d", helpers.CurlConnectTimeout)
+
+		kubectl          *helpers.Kubectl
+		microscopeCancel func() error
+		uptimeCancel     context.CancelFunc
+	)
+
+	BeforeAll(func() {
+		k8sVersion := helpers.GetCurrentK8SEnv()
+		switch k8sVersion {
+		case "1.7", "1.8":
+			Skip(fmt.Sprintf("Istio doesn't support K8S %s", k8sVersion))
+		}
+
+		kubectl = helpers.CreateKubectl(helpers.K8s1VMName(), logger)
+
+		err := kubectl.CiliumInstall(helpers.CiliumDefaultDSPatch, helpers.CiliumConfigMapPatch)
+		Expect(err).To(BeNil(), "Cilium cannot be installed")
+
+		ExpectCiliumReady(kubectl)
+		ExpectKubeDNSReady(kubectl)
+
+		By("Creating the istio-system namespace")
+		res := kubectl.NamespaceCreate(istioSystemNamespace)
+		res.ExpectSuccess("unable to create namespace %q", istioSystemNamespace)
+
+		By("Creating the Istio resources")
+		res = kubectl.Apply(istioYAMLPath)
+		res.ExpectSuccess("unable to create Istio resources")
+
+		// Ignore one-time jobs and Prometheus. All other pods in the
+		// namespaces have an "istio" label.
+		By("Waiting for Istio pods to be ready")
+		err = kubectl.WaitforPods(istioSystemNamespace, "-l istio", 300)
+		Expect(err).To(BeNil(),
+			"Istio pods are not ready after timeout in namespace %q", istioSystemNamespace)
+
+		for _, name := range istioServiceNames {
+			By("Waiting for Istio service %q to be ready", name)
+			err = kubectl.WaitForServiceEndpoints(
+				istioSystemNamespace, "", name, helpers.HelperTimeout)
+			Expect(err).Should(BeNil(), "Service %q is not ready after timeout", name)
+		}
+
+		for _, name := range istioServiceNames {
+			By("Waiting for DNS to resolve Istio service %q", name)
+			err = kubectl.WaitForKubeDNSEntry(name, istioSystemNamespace)
+			Expect(err).To(BeNil(), "DNS entry is not ready after timeout")
+		}
+	})
+
+	AfterAll(func() {
+		By("Deleting the Istio resources")
+		_ = kubectl.Delete(istioYAMLPath)
+
+		By("Deleting the istio-system namespace")
+		_ = kubectl.NamespaceDelete(istioSystemNamespace)
```

Comment:
Add a call to `WaitCleanAllTerminatingPods` to ensure that the test only finishes once all pods are deleted.

Reply:
Done.